app/uk/gov/nationalarchives/omega/editorial/controllers/EditSetRecordController.scala

```diff
@@ -309,11 +309,13 @@ class EditSetRecordController @Inject() (
     form: Form[EditSetRecordFormValues]
   )(implicit request: Request[AnyContent]): HtmlFormat.Appendable = {
     val title = resolvedMessage(MessageKeys.title)
+    val recordType = if (editSetRecord.oci.endsWith(".P")) resolvedMessage(MessageKeys.recordTypePhysical) else ""
```

Comment:
Do you think we should put this expression inside the view? It's just formatting, after all, and we're already passing the record in.

The advantage would be to cut down on the number of arguments we are passing into the view, which can get a bit unwieldy.

Reply:
@felixgb Yes can do, but the intention was to generated the type from the controller as I understand there could be possible changes needed if there are other types and it would bring the logic into the view

Reply:
I agree with Felix.

And it's probably not too early to introduce a `recordType` field to `EditSetRecord`; for now, it can be a derived, optional field in that it's value depends on the OCI suffix.

Reply:
For example, in `EditSetRecord`:

```
  sealed trait RecordType
  
  case object Physical extends RecordType
```

```
case class EditSetRecord(
  ccr: String,
  oci: String,
  scopeAndContent: String,
  coveringDates: String,
  formerReferenceDepartment: String,
  formerReferencePro: String,
  startDateDay: String,
  startDateMonth: String,
  startDateYear: String,
  endDateDay: String,
  endDateMonth: String,
  endDateYear: String,
  legalStatusID: String,
  placeOfDepositID: String,
  note: String,
  background: String,
  custodialHistory: String,
  relatedMaterial: Seq[RelatedMaterial] = Seq.empty,
  separatedMaterial: Seq[SeparatedMaterial] = Seq.empty,
  creatorIDs: Seq[String] = Seq.empty
) {

  val recordType: Option[RecordType] = if (oci.endsWith(".P")) Option(Physical) else None

}
```

Reply:
That solution looks good, both to keep the logic out of the view and out of the controller actions.

Reply:
Done

Reply:
Confirmed.